### PR TITLE
feat(helm): adding --name to update single repo

### LINKED ISSUE #5127: Feature: add --repo-name to 'helm repo update'
feature request
---------------

Output of `helm version`: v2.12.0
Output of `kubectl version`: v1.11.0

It would be grate to have `--repo-name` flag when using `helm repo update`
I think there are use cases, to update a specific repository, as sometimes one of the repositories update may fail for all kind of reasons, this is mainly intended to you with `--strict` flag

### PR DESCRIPTION
**Pull request by thesattiraju**
**What this PR does / why we need it**:
fixes https://github.com/helm/helm/issues/5127

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by ds-ms** [comment]
@bacongobbler / @adamreese  could you review this?

**Comment by adamreese** [comment]
Rather than adding a `--name` flag, could we just use arguments?  

Example: `helm repo update stable`

**Comment by ds-ms** [comment]
@adamreese I agree that removing `--name` makes it better.
We'll end up having one new path though right? i.e. `helm repo update` and `helm repo update <repo_name>`. 
I was skeptical about how to convey that `helm repo update` meant helm would update all repos, so I chose this path instead.

**Comment by ds-ms** [comment]
@adamreese I've updated it to `helm repo update <repo_name>` instead.

**Comment by ds-ms** [comment]
@adamreese @bacongobbler could you take a look at this PR?

**Comment by ds-ms** [comment]
@adamreese @bacongobbler ping

**Comment by ds-ms** [comment]
@adamreese @bacongobbler pinging again

**Review by hickeyma — CHANGES_REQUESTED** [review]
@ds-ms Thanks for the changes. It is nearly there but it needs some updates as follows:
- Do you mind updating the command `use` to include "[repo_name]" (https://github.com/helm/helm/blob/master/cmd/helm/repo_update.go#L56)
- Do you mind adding unit test to https://github.com/helm/helm/blob/master/cmd/helm/repo_update_test.go

**Comment by ds-ms** [comment]
@hickeyma Updated the PR

File: cmd/helm/repo_update_test.go [hunk 1]
```diff
@@ -132,3 +132,29 @@ func TestUpdateCmdStrictFlag(t *testing.T) {
 		t.Errorf("Expected 'Unable to get an update', got %q", got)
 	}
 }
+
```
Comment (hickeyma) [inline cmd/helm/repo_update_test.go]: Do you mind adding a test that tests for a successful update with a single repo name?

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks @ds-ms for the updates. I would appreciate if you could add a positive test as described in the inline comment.

File: cmd/helm/repo_update_test.go (hunk 1 above)
Comment (ds-ms) [reply cmd/helm/repo_update_test.go]: @hickeyma added.

File: cmd/helm/repo_update_test.go [hunk 2]
```diff
@@ -132,3 +132,78 @@ func TestUpdateCmdStrictFlag(t *testing.T) {
...
+	if err = cmd.RunE(cmd, []string{"randomRepo"}); err == nil {
+		t.Fatal("expected error due to wrong repo name")
+	}
+
+	if got := fmt.Sprintf("%v", err); !strings.Contains(got, "no repositories found matching the provided name. Verify if the repo exists") {
+		t.Errorf("Expected 'no repositories found matching the provided name. Verify if the repo exists', got %q", got)
+	}
+}
+
+func TestUpdateChart(t *testing.T) {
```
Comment (hickeyma) [inline cmd/helm/repo_update_test.go]: Should this test be named something like "TestUpdateRepo"?

File: cmd/helm/repo_update_test.go [hunk 3]
```diff
@@ -132,3 +132,78 @@ func TestUpdateCmdStrictFlag(t *testing.T) {
...
+		os.RemoveAll(thome.String())
+		cleanup()
+	}()
+	if err := ensureTestHome(hh, t); err != nil {
+		t.Fatal(err)
+	}
+
+	settings.Home = thome
+
+	if err := addRepository("chart1", ts.URL(), "", "", hh, "", "", "", true); err != nil {
```
Comment (hickeyma) [inline cmd/helm/repo_update_test.go]: :s/chart1/repo1 ?

File: cmd/helm/repo_update_test.go [hunk 4]
```diff
@@ -132,3 +132,78 @@ func TestUpdateCmdStrictFlag(t *testing.T) {
...
+		t.Error(err)
+	}
+
+	if err := addRepository("chart2", ts.URL(), "", "", hh, "", "", "", true); err != nil {
```
Comment (hickeyma) [inline cmd/helm/repo_update_test.go]: :s/chart2/repo2 ?

File: cmd/helm/repo_update_test.go [hunk 5]
```diff
@@ -132,3 +132,78 @@ func TestUpdateCmdStrictFlag(t *testing.T) {
...
+		t.Error(err)
+	}
+
+	out := bytes.NewBuffer(nil)
+	cmd := newRepoUpdateCmd(out)
+
+	if err = cmd.RunE(cmd, []string{"chart1"}); err != nil {
+		t.Fatal("expected to update newChart correctly")
```
Comment (hickeyma) [inline cmd/helm/repo_update_test.go]: :s/newChart/newRepo

File: cmd/helm/repo_update_test.go [hunk 6]
```diff
@@ -132,3 +132,78 @@ func TestUpdateCmdStrictFlag(t *testing.T) {
...
+	}
+
+	got := out.String()
+
+	if !strings.Contains(got, "Successfully got an update from the \"chart1\"") {
```
Comment (hickeyma) [inline cmd/helm/repo_update_test.go]: :s/chart1/repo1

File: cmd/helm/repo_update_test.go [hunk 7]
```diff
@@ -132,3 +132,78 @@ func TestUpdateCmdStrictFlag(t *testing.T) {
...
+		t.Errorf("Expected to successfully update \"chart1\" repository, got %q", got)
+	}
+
+	if strings.Contains(got, "Successfully got an update from the \"chart2\"") {
```
Comment (hickeyma) [inline cmd/helm/repo_update_test.go]: :s/chart2/repo2

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks for adding the positive test. Some nits on it. When you deliver changes, can you also squash the changes on this PR?

**Comment by ds-ms** [comment]
@hickeyma updated

File: cmd/helm/repo_update.go [hunk 8]
```diff
@@ -53,12 +62,15 @@ func newRepoUpdateCmd(out io.Writer) *cobra.Command {
 		update: updateCharts,
 	}
 	cmd := &cobra.Command{
-		Use:     "update",
+		Use:     "update [repo_name]",
```
Comment (hickeyma) [inline cmd/helm/repo_update.go]: :s/repo_name/REPO_NAME
for consistency with commands like `helm get`, `helm install` etc.

**Review by hickeyma — CHANGES_REQUESTED** [review]
Nearly there. Just small nit on caps on repo name for consistency.

File: cmd/helm/repo_update.go (hunk 8 above)
Comment (ds-ms) [reply cmd/helm/repo_update.go]: done

**Review by hickeyma — APPROVED** [review]
LGTM, thanks.

**Pull request merged**